partner_communication/models/communication_job.py

```diff
@@ -0,0 +1,177 @@
+     and to send again (or print again) a particular communication.
+
+     It is also useful to batch send communications without manually looking
+     for which one to send by e-mail and which one to print.
+     """
+    _name = 'partner.communication.job'
+    _description = 'Communication Job'
+    _order = 'date desc,sent_date desc'
+
+    ##########################################################################
+    #                                 FIELDS                                 #
+    ##########################################################################
+    config_id = fields.Many2one('partner.communication.config',
+                                'Type', required=True)
+    partner_id = fields.Many2one('res.partner', 'Send to', required=True)
+    object_id = fields.Integer('Resource id', required=True)
+
+    date = fields.Datetime(default=fields.Datetime.now())
+    sent_date = fields.Datetime()
+    state = fields.Selection([
+        ('pending', _('Pending')),
+        ('done', _('Done')),
+        ('cancel', _('Cancelled')),
+    ], default='pending', readonly=True)
+
+    auto_send = fields.Boolean(
+        help='Job is processed at creation if set to true')
+    send_mode = fields.Selection(
+        [('digital', _('By e-mail')), ('physical', _('Print report'))],
+        compute='_compute_send_mode', inverse='_inverse_generation',
+        store=True)
+    email_template_id = fields.Many2one(
+        related='config_id.email_template_id', store=True)
+    report_id = fields.Many2one(related='config_id.report_id', store=True)
+    from_employee_id = fields.Many2one(
+        'hr.employee', 'Communication From',
+        help='The sponsor will receive the communication from this employee'
+    )
+    email_to = fields.Char(
+        help='optional e-mail address to override recipient')
+    email_id = fields.Many2one('mail.mail', 'Generated e-mail')
+    generated_document = fields.Many2one('ir.attachment')
+
+    body_html = fields.Html(
+        compute='_compute_html', inverse='_inverse_generation',
+        store=True)
+
+    ##########################################################################
+    #                             FIELDS METHODS                             #
+    ##########################################################################
+    @api.depends('partner_id', 'config_id')
+    @api.multi
+    def _compute_send_mode(self):
+        for job in self:
+            if job.config_id:
+                job.send_mode = job.config_id.get_inform_mode(
+                    job.partner_id)[0]
+
+    @api.depends('object_id', 'email_template_id')
+    @api.multi
+    def _compute_html(self):
+        for job in self:
+            if job.object_id and job.email_template_id:
+                job.body_html = self.env[
+                    'mail.compose.message'].get_generated_html(
+                    job.email_template_id, [job.object_id])
+
+    @api.multi
+    def _inverse_generation(self):
```

Comment:
It is stupid but it's the way to allow writing in computed fields: we must define an `inverse` method which could do something when we update the field. In our case we have nothing to do when writing in these fields so we can just return true.

But if I remove the methode, the fields become readonly.